### ISSUE TITLE
% in LIKE is potentially not handled correctly (with line break)

### ISSUE THREAD

**Issue by XenoAmess** [ClickHouse/ClickHouse#46654]
LIKE expr behaves very very weird when meating \n and \\

https://fiddle.clickhouse.com/54c8d7c6-7201-4794-8609-f9c6125de079

```sql
CREATE TABLE MY_TABLE(xForwardedFor String) ENGINE=Memory;

INSERT INTO MY_TABLE(xForwardedFor) VALUES ( 'X-Forwarded-For: (null)\nCdn-Src-Ip: (null)\nClientip: (null)');

SELECT count(*)
FROM MY_TABLE
WHERE like(toString(`xForwardedFor`), '%X\\-Forwarded\\-For%');

SELECT count(*)
FROM MY_TABLE
WHERE like(toString(`xForwardedFor`), '%X\\-Forwarded\\-For%Cdn%');

SELECT count(*)
FROM MY_TABLE
WHERE like(toString(`xForwardedFor`), '%X-Forwarded-For%');

SELECT count(*)
FROM MY_TABLE
WHERE like(toString(`xForwardedFor`), '%X-Forwarded-For%Cdn%');

SELECT count(*)
FROM MY_TABLE
WHERE like(toString(`xForwardedFor`), '%X\\-Forwarded\\-For: \\(null\\)%\\\nCdn\\-Src\\-Ip: \\(null\\)%\\\nClientip: \\(null\\)%');

```

response

```
1
0
1
1
0
```

**Does it reproduce on recent release?**

yes

**Expected behavior**

response

```
1
1
1
1
1
```
**Additional context**

**Comment by XenoAmess** [comment]
even more confusings in https://fiddle.clickhouse.com/a48e19c2-8b55-49c1-909a-36f4b7314c8b

**Comment by rschu1ze** [comment]
I played with the examples a bit but could not really understand what the problem is. To me, it looks like these issues should be solved with #46869 and #46875. Would you like to check again once both PRs are merged and then create a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)? Thanks!

**Comment by XenoAmess** [comment]
> I played with the examples a bit but could not really understand what the problem is. To me, it looks like these issues should be solved with #46869 and #46875. Would you like to check again once both PRs are merged and then create a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)? Thanks!

Yep, would run all tests I mentioned in the several issues after the prs merged in and see if they becomes correct.
